keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -526,6 +578,9 @@ class ClusterHandler(
       "spinnaker:stack",
       "spinnaker:details"
     )
+
+    private val REGION_PLACEHOLDER = """\{\{region\}\}""".toRegex()
+    private val REGION_PATTERN = """([a-z]{2}(-gov)?)-([a-z]+)-\d"""
```

Comment:
Why is one of these done with `toRegex` and not the other? Also I don't understand the first one.

Reply:
These are used in this block:
https://github.com/spinnaker/keel/blob/3044fd1348b93f012e1e678247a322bad59fd202/keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt#L515-L529
If the default key pair configured in clouddriver includes the region placeholder, then we replace that with the pattern to create a new regex on-the-fly and compare against that particular server group's current config.

For example, if the default is `"nf-keypair-123456789012-{{region}}"` and the server group we're looking has a launch config with a key pair like `"nf-keypair-123456789012-us-west-2"`, it will match and we omit the key pair from the export.

Make sense?

P.S. I just realized I could/should have used `REGION_PLACEHOLDER` in that statement with the `replace()`. 😛

Reply:
If you're searching for the literal string `{{region}}` it also feels like a regex might be overkill ;-)

Reply:
Yeah, I had more stuff in there initially then ended up with just the region placeholder. Will make the change.